### PR TITLE
[upgrade] 25 - Upgrade eslint to latest version

### LINKED ISSUE #25: Update ESLint
Project is currently bound to ESLint `^1.10.3`, latest is `3.9.1`. There are breaking changes between versions, so updating will be more involved than binding latest in package.json.

### PR DESCRIPTION
**Pull request by ajfuller**
Closes #25 

👋 Hey guys! This PR updates eslint to the latest version along with adding two additional plugins that will help with linting.

- https://github.com/prettier/eslint-config-prettier
   - Disables eslint rules that are already taken care of by prettier
- https://github.com/mysticatea/eslint-plugin-node
   - Allows us to set the supported versions and disallows functionality that isn't available. For example, if it was set to 4.0 it would disallow `const` or destructuring.

Also disabled a few rules that came with the latest Airbnb config. Most notably `prefer-arrow-callback` since that would have required many changes across the app and `strict` because I could not get a configuration that would recognize that only commonjs modules are being used.

### CODE REVIEW

**Comment by AndersDJohnson** [comment]
😮 A wild Fuller appears! 👍

File: app/lib/CaptureRecorder.js [hunk 1]
```diff
@@ -18,7 +20,7 @@ function CaptureRecorder(app, captureName) {
   assert(captureName, 'Capture name required');
 
   this.app = app;
-  this.capturePath = path.normalize(app.config.capturesDir + '/' + captureName);
+  this.capturePath = path.normalize(app.config.capturesDir + '/' + captureName); // eslint-disable-line prefer-template
```
Comment (AndersDJohnson) [inline app/lib/CaptureRecorder.js]: Could just go with a template string here.

File: app/lib/CapturePlayer.js [hunk 2]
```diff
@@ -1,19 +1,19 @@
...
 /* eslint-disable no-process-exit, no-sync */
 
 const assert = require('assert');
 const fs = require('fs');
 const path = require('path');
+const url = require('url');
 
 function normalizeRewritePath(_path) {
-  return _path.replace(/[\?\=\&\%\+]+/g, '_').replace(/^\/?/, '/');
+  return _path.replace(/[\?\=\&\%\+]+/g, '_').replace(/^\/?/, '/'); // eslint-disable-line no-useless-escape
```
Comment (AndersDJohnson) [inline app/lib/CapturePlayer.js]: Any reason to keep the useless escapes, assuming they are actually useless?

File: app/lib/CaptureRecorder.js (hunk 1 above)
Comment (ajfuller) [reply app/lib/CaptureRecorder.js]: Definitely could. I can go back through and make those happen.

File: app/lib/CapturePlayer.js (hunk 2 above)
Comment (ajfuller) [reply app/lib/CapturePlayer.js]: Simplest explanation is I don't trust my RegEx enough to not somehow break it 😄

**Comment by AndersDJohnson** [comment]
@ryanricard You good with me merging this?

**Comment by AndersDJohnson** [comment]
@ajfuller As a side note, are you using mockyeah? Curious to learn about your use case.

**Comment by ajfuller** [comment]
@AndersDJohnson Not yet, but I'm proposing using it as a fixture server during development and for visual regression testing so we don't have to compile and run our backend code locally. Probably going to try integrating some time in the next week so I'll try to log any questions or feedback I have.

**Comment by AndersDJohnson** [comment]
@ajfuller Sweet. There are a lot of new undocumented features I've added, and more on the way. One major one is matching query parameters and request body. Another major one is a transparent proxy to real backend when there's no matching mock. Another is HTTPS support with automatic certificate generation if no cert file paths are provided in configuration. Let us know if you have any other ideas!

**Pull request merged**